### PR TITLE
Add support for storing and using default branch so we can get the right readme

### PR DESCRIPTION
**Pull request by halkeye**
Fixes https://github.com/jenkins-infra/plugin-site/issues/630 and a couple times reported on gitter.

I'm not proud of switching all the functions from strings to plugins, but as more and more data is needed to fetch readmes, it gets tricker to pass in single data.

### CODE REVIEW

**Review by timja — APPROVED** [review]

File: src/main/java/io/jenkins/plugins/models/Plugin.java [hunk 1]
```diff
@@ -19,6 +19,22 @@
...
+			return true;
+		// null check
+		if (o == null)
+			return false;
+		// type check and cast
+		if (getClass() != o.getClass())
+			return false;
+		Plugin plugin = (Plugin) o;
+		// field comparison
+		return gav.equals(plugin.gav) && getWikiUrl().equals(plugin.getWikiUrl());
```
[the remark this answers is not in the document]
Comment (halkeye) [reply src/main/java/io/jenkins/plugins/models/Plugin.java:35]: <_< lazyness for writing tests? the default gav for them is null so they would collide., but honestly its probably not needed anymore, it was needed before a bunch of refactoring. Lemme confirm.

[the remark this answers is not in the document]
Comment (halkeye) [reply src/main/java/io/jenkins/plugins/models/Plugin.java]: yea, no longer needed, fixed.

File: src/test/java/io/jenkins/plugins/services/WikiServiceTest.java [hunk 2]
```diff
@@ -53,7 +56,10 @@ public void testGetWikiContentGitCustom() {
 
   public void testGetWikiContentGit(String url) {
     System.setProperty("github.client.id", "dummy");
-    final String content = wikiService.getWikiContent(url);
+    final String content = wikiService.getWikiContent(new Plugin() {
+      @Override public String getWikiUrl() { return url; }
+      @Override public String getDefaultBranch() { return "master"; }
```
Comment (daniel-beck) [inline src/test/java/io/jenkins/plugins/services/WikiServiceTest.java:61]: Perhaps it's a better look to use 
```suggestion
      @Override public String getDefaultBranch() { return "main"; }
```
here (that's the new GH default anyway) + below, and to use a sample plugin that uses a `main` default branch?

**Review by daniel-beck — APPROVED** [review]
🎉

**Pull request merged**